File: problem/Equal_Operations_II/canBeEqual.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
const checkStrings = (s1, s2) => {
    const freq = Array(52).fill(0);

    for (let i = 0; i < s1.length; i++) {
        const off = (i & 1) * 26;
        freq[s1.charCodeAt(i) - 97 + off]++;
        freq[s2.charCodeAt(i) - 97 + off]--;
    }

    return freq.every(c => c === 0);
};

File: problem/Equal_Operations_II/canBeEqual.c
bool checkStrings(char *s1, char *s2)
{
    int freq[52] = {0};
    int n = strlen(s1);

    for (int i = 0; i < n; i++)
    {
        int off = (i & 1) * 26;
        freq[s1[i] - 'a' + off]++;
        freq[s2[i] - 'a' + off]--;
    }

    for (int i = 0; i < 52; i++)
    {
        if (freq[i] != 0)
        {
            return false;
        }
    }

    return true;
}